tracer/src/Datadog.Trace/Configuration/Schema/NamingSchema.cs

```diff
@@ -33,5 +45,25 @@ public NamingSchema(SchemaVersion version, bool peerServiceTagsEnabled, bool rem
         public ServerSchema Server { get; }
 
         public bool RemoveClientServiceNamesEnabled { get; }
+
+        public void RemapPeerService(ITags tags)
+        {
+            if (!_peerServiceTagsEnabled || _peerServiceNameMappings is null || _peerServiceNameMappings.Count == 0)
+            {
+                return;
+            }
```

Comment:
Looking at how `DD_TRACE_PEER_SERVICE_DEFAULTS_ENABLED` should work, it looks like it [should be ignored](https://github.com/DataDog/dd-trace-dotnet/blob/8990da00dceff362df36bb8a0c41691f5f228f00/tracer/src/Datadog.Trace/Configuration/ConfigurationKeys.cs#L421) when on v1

> Configuration key for automatically populating the peer.service tag
> from predefined precursor attributes when the span attribute schema is v0.
> This is ignored when the span attribute schema is v1 or later.
> Default value is false

So maybe we should do something like this?

```suggestion
            if ((Version.Equals(SchemaVersion.V0) && !_peerServiceTagsEnabled) || _peerServiceNameMappings is null || _peerServiceNameMappings.Count == 0)
            {
                return;
            }
```

Reply:
BTW, otherwise everything LGTM

Reply:
Thanks a lot, indeed I completely missed it. 
I wonder if we should do things differently (like for it to true in the settings when schema is set to 1 to avoid forgetting but that's a broader change so your suggestion is perfect.